tests/e2e/objects/views/ResourcesView.ts

```diff
@@ -113,26 +118,40 @@ export class ResourcesView extends View {
    */
   get ccloudSchemaRegistries(): Locator {
     // third nested element: Confluent Cloud item -> environment item -> Schema Registry item
-    return this.body
-      .locator("[role='treeitem'][aria-level='3']")
-      .filter({ has: this.page.locator(".codicon-confluent-schema-registry") });
+    return this.schemaRegistries.and(
+      this.page.locator("[aria-level='3'][aria-label^='CCLOUD connection: Schema Registry']"),
+    );
   }
 
   /**
    * Locator for local Schema Registry tree items.
    * Only visible when the {@link localItem "Local" item} is expanded.
    */
   get localSchemaRegistries(): Locator {
-    return this.schemaRegistries.filter({ hasText: "local-sr" });
+    // second nested element: Local connection item -> Schema Registry item
+    return this.schemaRegistries.and(
+      this.page.locator("[aria-level='2'][aria-label^='LOCAL connection: Schema Registry']"),
+    );
   }
 
   /**
    * Locator for direct connection Schema Registry tree items.
-   * Only visible when a {@link directConnections "Direct Connections" item} is available and
-   * expanded.
+   * Only visible when a {@link directConnections direct connection item} is available and
+   * expanded with a Schema Registry configured.
    */
   get directSchemaRegistries(): Locator {
-    return this.schemaRegistries.filter({ hasText: "Schema Registry" });
+    // second nested element: Direct connection item -> Schema Registry item
+    return this.schemaRegistries.and(
+      this.page.locator("[aria-level='2'][aria-label^='DIRECT connection: Schema Registry']"),
+    );
+  }
+
+  /** Locator for all Flink Compute Pool tree items.*/
+  get flinkComputePools(): Locator {
+    // only available for CCloud connections
+    return this.treeItems.filter({
+      has: this.page.locator(".codicon-confluent-flink-compute-pool"),
+    });
```

Comment:
This one wasn't strictly necessary since we won't have local/direct compute pools anytime soon, but it keeps with the existing `<resource>` -> `<connectionType><resource>` getter patterns.